src/gash/router/server/CommandChannelHandler.java

```diff
@@ -117,7 +184,55 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws E
 		ctx.close();
 	}
 
-	public RoutingConf getConf()    {
+	public static class ClientClosedListener implements ChannelFutureListener {
+		CommandChannelHandler cc;
+
+		public ClientClosedListener(CommandChannelHandler commandChannelHandler) {
+			this.cc = commandChannelHandler;
+		}
+
+		@Override
+		public void operationComplete(ChannelFuture future) throws Exception {
+			// we lost the connection or have shutdown.
+			System.out.println("--> client lost connection to the server");
+			System.out.flush();
+			boolean result=channelMap.remove(future.channel().remoteAddress(),future.channel());
+			if(!result){
+				System.out.println("Error while deleting entry from Hash Map");
+			}	
+		}
+	}
+
+	public RoutingConf getConf() {
 		return conf;
 	}
+
+	public Channel connect() {
+		if (channel == null) {
+			init();
```

Comment:
This method is creating new eventloop. What if earlier event loop is running. Basically, event loop is a new thread. We need to shut it down gracefully.